### PR TITLE
Normalization handles quote in column names 

### LINKED ISSUE #4729: [MSSQL -> Snowflake] Normalization issue with ' in columns names
## Enviroment
- **Airbyte version**:  0.27.2-alpha
- **OS Version / Instance**: GCP COS
- **Deployment**: Docker
- **Source Connector and version**: MSSQL 0.3.2
- **Destination Connector and version**: Snowflake 0.3.10
- **Severity**: High
- **Step where error happened**: Reset

## Current Behavior
It seems like the issue could be link to wrongfully prepared statements, thus SQL injection. The first syntax error appears to be `syntax error line 34 at position 65 unexpected 't'` which indicates that the value is probably something like `don't`. The single quote is not escaped in the database resulting in an SQL error during normalization.

## Expected Behavior
The normalization process should be handled with proper prepared statements and not string injection inside the SQL queries.

## Logs

<details>
<summary>LOG</summary> 

```
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 - [31mCompleted with 2 errors and 0 warnings:[0m
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 - 
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 - [33mDatabase Error in model <REDACTED_TABLE_NAME_1> (models/generated/airbyte_tables/SAP_RAW_REF/<REDACTED_TABLE_NAME_1>.sql)[0m
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   001003 (42000): SQL compilation error:
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 34 at position 65 unexpected 't'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 34 at position 67 unexpected 'Match'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 109 at position 19 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 111 at position 15 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 112 at position 19 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 114 at position 15 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 118 at position 19 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 120 at position 15 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 121 at position 19 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 123 at position 15 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 184 at position 41 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 186 at position 41 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 190 at position 41 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 192 at position 41 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 229 at position 14 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 230 at position 14 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 232 at position 14 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 233 at position 14 unexpected '02'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   parse error line 240 at position 8 near '<EOF>'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_tables/SAP_RAW_REF/<REDACTED_TABLE_NAME_1>.sql
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 - 
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 - [33mDatabase Error in model <REDACTED_TABLE_NAME_2> (models/generated/airbyte_tables/SAP_RAW_REF/<REDACTED_TABLE_NAME_2>.sql)[0m
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   001003 (42000): SQL compilation error:
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 7 at position 58 unexpected '17'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 7 at position 69 unexpected 'Apr'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   syntax error line 9 at position 63 unexpected ')'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   parse error line 156 at position 8 near '<EOF>'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   parse error line 156 at position 8 near '<EOF>'.
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 -   compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_tables/SAP_RAW_REF/<REDACTED_TABLE_NAME_2>.sql
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 - 
2021-07-13 13:53:46 INFO () LineGobbler(voidCall):85 - Done. PASS=22 WARN=0 ERROR=2 SKIP=0 TOTAL=24
2021-07-13 13:53:46 INFO () TemporalAttemptExecution(lambda$getWorkerThread$2):149 - Completing future exceptionally...
io.airbyte.workers.WorkerException: Normalization Failed.
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:69) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:36) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:146) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: io.airbyte.workers.WorkerException: Normalization Failed.
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:66) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	... 3 more
	Suppressed: io.airbyte.workers.WorkerException: Normalization process wasn't successful
		at io.airbyte.workers.normalization.DefaultNormalizationRunner.close(DefaultNormalizationRunner.java:151) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
		at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:60) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
		at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:36) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
		at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:146) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
		at java.lang.Thread.run(Thread.java:832) [?:?]
2021-07-13 13:53:46 INFO () TemporalAttemptExecution(get):133 - Stopping cancellation check scheduling...
2021-07-13 13:53:46 WARN () POJOActivityTaskHandler$POJOActivityImplementation(execute):243 - Activity failure. ActivityId=6c954a2a-f6f3-3c42-ae90-5fba5637ec71, activityType=Normalize, attempt=1
java.util.concurrent.ExecutionException: io.airbyte.workers.WorkerException: Normalization Failed.
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2063) ~[?:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.get(TemporalAttemptExecution.java:131) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	at io.airbyte.workers.temporal.SyncWorkflow$NormalizationActivityImpl.normalize(SyncWorkflow.java:321) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:564) ~[?:?]
	at io.temporal.internal.sync.POJOActivityTaskHandler$POJOActivityInboundCallsInterceptor.execute(POJOActivityTaskHandler.java:277) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.sync.POJOActivityTaskHandler$POJOActivityImplementation.execute(POJOActivityTaskHandler.java:216) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.sync.POJOActivityTaskHandler.handle(POJOActivityTaskHandler.java:181) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:192) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:154) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.worker.PollTaskExecutor.lambda$process$0(PollTaskExecutor.java:73) ~[temporal-sdk-1.0.4.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: io.airbyte.workers.WorkerException: Normalization Failed.
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:69) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:36) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:146) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	... 1 more
Caused by: io.airbyte.workers.WorkerException: Normalization Failed.
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:66) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:36) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:146) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
	... 1 more
	Suppressed: io.airbyte.workers.WorkerException: Normalization process wasn't successful
		at io.airbyte.workers.normalization.DefaultNormalizationRunner.close(DefaultNormalizationRunner.java:151) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
		at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:60) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
		at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:36) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
		at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:146) ~[io.airbyte-airbyte-workers-0.27.2-alpha.jar:?]
		at java.lang.Thread.run(Thread.java:832) [?:?]
```

</details>

## Steps to Reproduce
1. Have data containing non escaped single quote in MSSQL
2. Reset the data with the appropriate tables from MSSQL to Snowflake

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
This PR depends on some changes made in bigquery/mysql destinations: https://github.com/airbytehq/airbyte/pull/5026

Closes https://github.com/airbytehq/airbyte/issues/4729

To avoid noise, output test files are regenerated and part of another PR https://github.com/airbytehq/airbyte/pull/5028

## How
Parse JSON blob while managing quote characters depending on the destination

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/bigquery/test_primary_key_streams/final/airbyte_ctes/test_normalization/nested_stream_with_complex_columns_resulting_into_long_names_partition_ab1.sql [hunk 1]
```diff
@@ -8,6 +8,7 @@ select
     _airbyte_nested_stream_with_complex_columns_resulting_into_long_names_hashid,
     json_extract_array(`partition`, "$['double_array_data']") as double_array_data,
     json_extract_array(`partition`, "$['DATA']") as DATA,
+    json_extract_array(`partition`, "$['column___with__quotes']") as column___with__quotes,
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/bigquery/test_primary_key_streams/final/airbyte_ctes/test_normalization/nested_stream_with_complex_columns_resulting_into_long_names_partition_ab1.sql:11]: due to PR https://github.com/airbytehq/airbyte/pull/5026, the raw json blob also contains some sanitized columns that we can extract from where quote characters have already been replaced by `_`

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mysql/test_primary_key_streams/final/airbyte_ctes/test_normalization/nested_stream_with_co__g_names_partition_ab1.sql [hunk 2]
```diff
@@ -5,9 +5,11 @@
 select
     _airbyte_nested_strea__nto_long_names_hashid,
     json_extract(`partition`, 
-  '$."double_array_data"') as double_array_data,
+    '$."double_array_data"') as double_array_data,
     json_extract(`partition`, 
-  '$."DATA"') as `DATA`,
+    '$."DATA"') as `DATA`,
+    json_extract(`partition`, 
+    '$."column___with__quotes"') as `column__'with"_quotes`,
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mysql/test_primary_key_streams/final/airbyte_ctes/test_normalization/nested_stream_with_co__g_names_partition_ab1.sql:12]: due to PR https://github.com/airbytehq/airbyte/pull/5026, the raw json blob in MySql also contains some sanitized columns that we can extract from where quote characters have already been replaced by `_`

Notice that the json_extract can't handle quotes in the json path but it is possible to create column names using `'` or `"` in MySQL...

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_primary_key_streams/final/airbyte_ctes/test_normalization/nested_stream_with_c___long_names_partition_ab1.sql [hunk 3]
```diff
@@ -6,6 +6,7 @@ select
     _airbyte_nested_stre__nto_long_names_hashid,
     jsonb_extract_path("partition", 'double_array_data') as double_array_data,
     jsonb_extract_path("partition", 'DATA') as "DATA",
+    jsonb_extract_path("partition", 'column`_''with"_quotes') as "column`_'with""_quotes",
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_primary_key_streams/final/airbyte_ctes/test_normalization/nested_stream_with_c___long_names_partition_ab1.sql:9]: Postgres is able to parse the json blob just fine. It just has an edge case of doubling the `'` characters to "escape" it...

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/snowflake/test_primary_key_streams/final/airbyte_ctes/TEST_NORMALIZATION/NESTED_STREAM_WITH_COMPLEX_COLUMNS_RESULTING_INTO_LONG_NAMES_PARTITION_AB1.sql [hunk 4]
```diff
@@ -6,6 +6,7 @@ select
     _AIRBYTE_NESTED_STREAM_WITH_COMPLEX_COLUMNS_RESULTING_INTO_LONG_NAMES_HASHID,
     get_path(parse_json(PARTITION), '"double_array_data"') as DOUBLE_ARRAY_DATA,
     get_path(parse_json(PARTITION), '"DATA"') as DATA,
+    get_path(parse_json(PARTITION), '"column`_''with""_quotes"') as "column`_'with""_quotes",
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/snowflake/test_primary_key_streams/final/airbyte_ctes/TEST_NORMALIZATION/NESTED_STREAM_WITH_COMPLEX_COLUMNS_RESULTING_INTO_LONG_NAMES_PARTITION_AB1.sql:9]: Snowflake is able to parse the json blob just fine. It just has an edge case of doubling the `'` or '"' characters to "escape" them...

**Review by tuliren — APPROVED** [review]

**Pull request merged**